open_spiel/algorithms/infostate_tree.cc

```diff
@@ -491,12 +491,57 @@ std::shared_ptr<InfostateTree> MakeInfostateTree(
     const Game& game, Player acting_player,
     int max_move_limit) {
   return std::shared_ptr<InfostateTree>(new InfostateTree(
-      game, acting_player, max_move_limit));
+      {game.NewInitialState().get()}, /*chance_reach_probs=*/{1.},
+      game.MakeObserver(kInfoStateObsType, {}),
+      acting_player, max_move_limit));
+}
+
+std::shared_ptr<InfostateTree> MakeInfostateTree(
+    const std::vector<InfostateNode*>& start_nodes,
+    int max_move_ahead_limit) {
+  std::vector<const InfostateNode*> const_nodes(
+      start_nodes.begin(), start_nodes.end());
+  return MakeInfostateTree(const_nodes, max_move_ahead_limit);
+}
+
+std::shared_ptr<InfostateTree> MakeInfostateTree(
+    const std::vector<const InfostateNode*>& start_nodes,
+    int max_move_ahead_limit) {
+  SPIEL_CHECK_FALSE(start_nodes.empty());
+  const InfostateNode* some_node = start_nodes[0];
+  const InfostateTree& originating_tree = some_node->tree();
+  SPIEL_DCHECK_TRUE([&](){
+    for (const InfostateNode* node : start_nodes) {
+      if (!node) return false;
+      if (!node->is_leaf_node()) return false;
+      if (node->depth() != some_node->depth()) return false;
+      if (&node->tree() != &originating_tree) return false;
+    }
+    return true;
+  }());
+
+  std::vector<const State*> start_states;
+  start_states.reserve(start_nodes.size() * 8);
```

Comment:
What is 8? Shouldn't it be a `constexpr` value in some class? If it's just a guess how much space will be most likely enough (to avoid reallocations), then it's okay but maybe a comment would help.